src/java.base/share/classes/sun/security/provider/SHA3.java

```diff
@@ -91,10 +95,12 @@ void implCompress(byte[] b, int ofs) {
 
     @IntrinsicCandidate
     private void implCompress0(byte[] b, int ofs) {
-       for (int i = 0; i < buffer.length; i++) {
-           state[i] ^= b[ofs++];
-       }
-       keccak();
+        b2lLittle(b, ofs, longBuf, 0, blockSize);
```

Comment:
What about BigEndian machines?

Reply:
The byte array should be interpreted as a little endian long array. b2lLittle() does just that on both little and big endian machines.

Reply:
Okay.